### PR TITLE
Handle json syntax errors in file imports.

### PR DESCRIPTION
**Pull request by kenzieschmoll**

### CODE REVIEW

File: packages/devtools/lib/src/framework/framework.dart [hunk 1]
```diff
@@ -83,50 +83,49 @@ class Framework {
...
+              ' DevTools screen: $devToolsScreen.',
+              hideDelay: Toast.extendedHideDelay,
+            );
+        }
+      } on FormatException catch (e) {
         toast(
-          'The imported file is not a Dart DevTools file. At this time, '
-          'DevTools only supports importing files that were originally exported'
-          ' from DevTools.',
+          'SyntaxError in imported file: "$e". Please make sure the imported '
```
Comment (jacob314) [inline packages/devtools/lib/src/framework/framework.dart]: SyntaxError in --> JSON syntax error in
?

File: packages/devtools/lib/src/framework/framework.dart [hunk 2]
```diff
@@ -83,50 +83,49 @@ class Framework {
...
-      if (devToolsScreen == null) {
+        // TODO(jacobr): add the inspector handling case here once the inspector
+        // can be exported.
+        switch (devToolsScreen) {
+          case timelineScreenId:
+            _importTimeline(import);
+            break;
+          default:
+            toast(
+              'Could not import file. The imported file is from an unrecognized'
```
Comment (jacob314) [inline packages/devtools/lib/src/framework/framework.dart]: Maybe change message to:
```
Could not import file.
The imported file is from '$devToolScreen' which is not supported by this version of Dart DevTools.
You may need to upgrade your version of Dart DevTools to view this file.
```

File: packages/devtools/lib/src/framework/framework.dart [hunk 3]
```diff
@@ -136,6 +135,25 @@ class Framework {
...
+    if (timelineScreen == null) {
+      addScreen(timelineScreen = TimelineScreen(disabled: false));
+    }
+    _enterImportMode();
+    navigateTo(timelineScreenId);
+    timelineScreen.loadFromImport(import);
+  }
+
+  void _enterImportMode() {
+    connectDialog.hide();
```
Comment (jacob314) [inline packages/devtools/lib/src/framework/framework.dart]: we should have a single helper that hides the connectDialog and the import message. I think we are doing that multiple places now.

File: packages/devtools/lib/src/framework/framework.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/framework/framework.dart]: done

File: packages/devtools/lib/src/framework/framework.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/framework/framework.dart]: done,.

File: packages/devtools/lib/src/framework/framework.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/framework/framework.dart]: This is the only place both are called now.

**Comment by jacob314** [comment]
lgtm

**Pull request merged**